aip/general/0121.md

```diff
@@ -61,6 +61,9 @@ exposes a large number of resources with a small number of methods on each
 resource. The methods can be either the standard methods ([Get][], [List][],
 [Create][], [Update][], [Delete][]), or [custom methods][].
 
+A resource that supports a [Create][], [Update][], or [Delete][] **must** also
```

Comment:
Is there any case where you would call something a resource that can `List` only? That would require returning repeated resources, which each have a name in that collection by definition.

At that point, you could say a resource **must** support `Get` and a resource collection **must** support `List` by definition - the mutable methods wouldn't need to be considered the reason.

Reply:
It sounds like the question is whether we can just mandate that all resources must support Get?

All of the examples I'm aware of that don't currently support Get (cloudsql users, tag bindings, bigquery reservation assignments, etc.) support List and at least one mutation.

Reply:
yeah, there's probably no good reason a *resource* API wouldn't support Get. It's a bit stronger of an assertion and it's harder to get approval the stronger the assertion, but I'll start there. 

Do we want to require List too? It seems better from a UX standpoint so you can retrieve a list of all resources you might want to operate on. But I could also imagine for security purposes you may not want to be able to enumerate all resources.

Although in the security case, you can control that ability with a permissions model.

Reply:
List: Good point. I wouldn't speculatively weaken the standard. *Must* implement List SGTM as a reasonable starting point.

Reply:
we could also go **must** for Get and **should** for List: i think the latter is more critical for uniform resources. 

But I'll change the PR for **must** to both and let's get some reviewers to chime in.

Reply:
updated, but I'll leave the thread open for discussion.